Scripts/Pointer/BasePointer.cs

```diff
@@ -0,0 +1,298 @@
+        /// </summary>
+        public PointerUnityEvent Selected = new PointerUnityEvent();
+
+        public abstract bool IsVisible();
+
+        /// <summary>
+        /// Reports the active state of the pointer.
+        /// </summary>
+        public bool Active
+        {
+            get;
+            protected set;
+        }
+
+        /// <summary>
+        /// Reports the valid state of the pointer.
+        /// </summary>
+        public bool Valid
+        {
+            get;
+            protected set;
+        }
+
+        /// <summary>
+        /// Reports hover duration of the pointer over the current target.
+        /// </summary>
+        public float HoverDuration
+        {
+            get;
+            protected set;
+        }
+
+        /// <summary>
+        /// Stores the RayCastHit information of the current pointer collision.
+        /// </summary>
+        public RaycastHit CollisionData
+        {
+            get;
+            protected set;
+        }
+
+        /// <summary>
+        /// Stores the Transform information of the cursor state.
+        /// </summary>
+        public Transform CursorData
+        {
+            protected get;
+            set;
+        }
+
+        protected RaycastHit cachedCollision;
+
+        /// <summary>
+        /// The Activate method turns on the pointer.
+        /// </summary>
+        public virtual void Activate()
+        {
+            Active = true;
+            OnActivated(GetPayload(CursorData, CollisionData));
+        }
+
+        /// <summary>
+        /// The Deactivate method turns off the pointer.
+        /// </summary>
+        public virtual void Deactivate()
+        {
+            Active = false;
+            TryEmitExit();
+            OnDeactivated(GetPayload(CursorData, cachedCollision));
+        }
+
+        /// <summary>
+        /// The Select method emits the selected event if the pointer is active and hovering over a valid target.
+        /// </summary>
+        public virtual void Select()
+        {
+            if (Active && Valid)
+            {
+                OnSelected(GetPayload(CursorData, CollisionData));
+            }
+        }
+
+        /// <summary>
+        /// The IsHovering method returns whether the pointer is currently hovering over a target.
+        /// </summary>
+        /// <returns>Returns `true` if the pointer is currently hovering over a target.</returns>
+        public virtual bool IsHovering()
+        {
+            return (HoverDuration > 0f);
+        }
+
+        /// <summary>
+        /// The IsElementVisible method determines if a given VisibilityType parameter should be considered visible.
+        /// </summary>
+        /// <param name="elementType">The VisibilityType to check for.</param>
+        /// <returns>Returns `true` if the VisibilityType given is considered visible.</returns>
+        public virtual bool IsElementVisible(VisibilityType elementType)
+        {
+            if (!enabled)

```

Comment:
I'm not sure if this should be checking against `enabled` or `isActiveAndEnabled` instead.

Reply:
I "think" this doesn't use activeandenabled because the actual valid/invalid gameobjects can be disabled.

but then this is actually on the gameobject of the script not the child elements so you may be right. i'll change it and see what breaks